### PR TITLE
data/selinux: update the policy to allow snapd to talk to org.freedesktop.timedate1

### PR DESCRIPTION
**Pull request by bboozzoo**
Snapd needs to poke the org.freedesktop.timedate1 service to find out whether
NTP was synchronized. That DBus API is provide by systemd-timesyncd (most
systems) or timedatex (CentOS mostly). The SELinux policy does not currently
allow talking to either service, so upon startup snapd will enter a deadlock
after getting blocked in the timeutil.IsNTPSynchronized() waiting for dbus
messages, while this is called in a code path that acquired state.Lock(), thus
blocking all Ensure() calls and interaction through the snapd socket.

Extend the SELinux policy to allow dbus communication to either timedate1
provider service.

### CODE REVIEW

**Review by stolowski — APPROVED** [review]
LGTM. Should we also look at making IsNTPSynchronized more robust and fail gracefully in such case, or is there nothing we can do about it?

**Comment by bboozzoo** [comment]
> LGTM. Should we also look at making IsNTPSynchronized more robust and fail gracefully in such case, or is there nothing we can do about it?

The catch is there was no error at all. The code just got stuck in go-dbus waiting for a reply that never came.

**Comment by stolowski** [comment]
> > LGTM. Should we also look at making IsNTPSynchronized more robust and fail gracefully in such case, or is there nothing we can do about it?
> 
> The catch is there was no error at all. The code just got stuck in go-dbus waiting for a reply that never came.

Is it "stuck hard", or would a timer + context cancellation handle it?

**Review by mvo5 — APPROVED** [review]
Thank you

**Comment by bboozzoo** [comment]
the test in question is green now, but there were some unrelated failures in the spread run anyway. @mvo5 can you land this branch?

**Comment by mvo5** [comment]
@bboozzoo Given that this is an interface change we either needs a +1 from @pedronis or @alexmurray - I hope (given how small/straightforward this is) that it's a simple one :)

**Pull request merged**
